### PR TITLE
Add review stacks changes

### PR DESCRIPTION
**Pull request by casperisfine**
Ref: https://github.com/Shopify/shipit-engine/pull/1102

Some changes after a first round of review.

cc @indiebrain

### CODE REVIEW

File: app/views/shipit/repositories/settings.html.erb [hunk 1]
```diff
@@ -8,28 +8,37 @@
...
+              When "Allow All", the provisioning label has no effect on dynamic stack provisioning - ALL Pull Requests dynamically provision stacks.
+            </small>
+          </p>
+          <p>
+            <small class="form-text text-muted" id="provisioningBehaviorHelp">
+              When "Allow With Label", dynamic provisioning occurs ONLY for Pull Requests whose labels include the 'Provisioning Label'.
+            </small>
+          </p>
+          <p>
+            <small class="form-text text-muted" id="provisioningBehaviorHelp">
```
Comment (benlangfeld) [inline app/views/shipit/repositories/settings.html.erb:32]: How come you went with multiple elements with the same ID?

Comment (casperisfine) [reply app/views/shipit/repositories/settings.html.erb:32]: Arf, I just did a nasty copy paste, and didn't pay attention... I don't think they need an id anyway.

Comment (casperisfine) [reply app/views/shipit/repositories/settings.html.erb]: I just pushed a fix, it's really not ideal either. To be honest frontend isn't exactly my forte so if you have suggestions, I'll welcome them.

**Review by indiebrain — APPROVED** [review]
Thanks for the cleanup pass. 📦 🚀 🤖

File: app/views/shipit/repositories/settings.html.erb [hunk 2]
```diff
@@ -8,28 +8,37 @@
 
     <div class="setting-section">
       <%= form_for @repository do |f| %>
-        <fieldset class="form-group">
```
[the remark this answers is not in the document]
Comment (casperisfine) [reply app/views/shipit/repositories/settings.html.erb:11]: I know, but it was styled quite badly, and I'm no good in CSS.

**Pull request merged**